homeassistant/components/switch/pilight.py

```diff
@@ -0,0 +1,216 @@
+"""
+
+Support for Pilight Switches.
+"""
+import logging
+import socket
+import json
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.helpers.entity import Entity
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = []
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+	""" Sets up a pilight socket client """
+	#_LOGGER.info(config)
+	host = config['ip']
+	port = config['port']
+	server = Pilight_Server(host, port)
```

Comment:
No, it seems that I was not clear enough. I meant not to check the variables themselves but if the establishing of the connection was successful. If the variables are wrong it would not be possible to create the connection anyway and the platform is unusable. 

Have a look at existing platforms ([here](https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/sensor/glances.py#L50) or [here](https://github.com/balloob/home-assistant/blob/dev/homeassistant/components/sensor/transmission.py#L36)).